### PR TITLE
synergy: 1.13.1.41 -> 1.14.1.32

### PR DESCRIPTION
**Pull request by osintowl**
###### Things done

- Built on platform(s)
  - [ x] x86_64-linux
  - [ x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/synergy/default.nix [hunk 1]
```diff
@@ -24,7 +24,7 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = lib.optional (!withGUI) "-DSYNERGY_BUILD_LEGACY_GUI=OFF";
```
Comment (jonringer) [inline pkgs/applications/misc/synergy/default.nix:25]: looks like this has changed:
```
CMake Warning at src/lib/arch/CMakeLists.txt:46 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake
```

Comment (jonringer) [reply pkgs/applications/misc/synergy/default.nix:25]:
```
https://github.com/NixOS/nixpkgs/pull/144267

1 package failed to build:
synergyWithoutGUI

2 packages built:
quicksynergy synergy
```

Comment (5A4B48) [reply pkgs/applications/misc/synergy/default.nix:25]: i'll see what I can do here, i'm amazed at everything you all do! Thank you for putting up with me!

File: pkgs/applications/misc/synergy/default.nix [hunk 2]
```diff
@@ -24,7 +24,7 @@ stdenv.mkDerivation rec {
...
 
-  nativeBuildInputs = [ cmake ] ++ lib.optional withGUI wrapQtAppsHook;
+  nativeBuildInputs = [ cmake pkg-config ] ++ lib.optional withGUI wrapQtAppsHook;
```
Comment (5A4B48) [inline pkgs/applications/misc/synergy/default.nix:27]: was I incorrect in adding pkg-config here and is that what is potentially breaking it? How do I test building the without gui version from source with my changes?

File: pkgs/applications/misc/synergy/default.nix (hunk 1 above)
Comment (5A4B48) [reply pkgs/applications/misc/synergy/default.nix:25]: I added a question on the nativeBuildInputs section, does adding pkg-config there break this? Net sure where or how this is failing.

File: pkgs/applications/misc/synergy/default.nix (hunk 2 above)
Comment (talyz) [reply pkgs/applications/misc/synergy/default.nix:27]: No, that's not what's breaking it. Upstream changes made Qt a required dependency whether the GUI is built or not, so the solution is to always add `wrapQtAppsHook` to the list.

**Comment by talyz** [comment]
I looked into upgrading synergy and while doing so for Linux/NixOS was pretty straight forward, macOS/darwin was a different story. I felt like it put this upgrade into "not very beginner friendly" territory, so I've opened up #145066 to replace this PR.

**Comment by jonringer** [comment]
#145066 was merged

Thanks @5A4B48 for your contribution :)

File: pkgs/applications/misc/synergy/default.nix (hunk 2 above)
Comment (5A4B48) [reply pkgs/applications/misc/synergy/default.nix:27]: Thank you!

**Pull request closed without merging**